uint8 sequence
uint8 sender
uint8 signal